pkg/apis/operator/v1alpha1/types_garden.go

```diff
@@ -764,6 +767,8 @@ type Credentials struct {
 	// Rotation contains information about the credential rotations.
 	// +optional
 	Rotation *CredentialsRotation `json:"rotation,omitempty"`
+	// EncryptionAtRest contains information about garden data encryption at rest.
+	EncryptionAtRest EncryptionAtRest `json:"encryptionAtRest"`
```

Comment:
Does this have to be a required field? Maybe due to potential future development? Currently it would be presented as an empty structure in yaml/json

Same for the shoot api

```suggestion
	// EncryptionAtRest contains information about garden data encryption at rest.
	// +optional
	EncryptionAtRest EncryptionAtRest `json:"encryptionAtRest,omitempty"`
```

Additionally, for new shoots the status will be created with default values, and the `EncryptionAtRest` field will be populated only later on (even when it contains other fields, apart from the encrypted resources).

Reply:
> Does this have to be a required field? Maybe due to potential future development?

This field has been created as required for potential future development. The plan is to add encryption provider metadata here as well, which should always be included.

Reply:
I still think that this should be optional and the actual child field that is required, be required (following what is done for other fields in the status).
So you would have something like `.status.credentials.encryptionAtRest.provider.type` that is required and `.status.credentials.encryptionAtRest.provider.configHash` (note that we might change `type` and `configHash` to what's necessary once work on that is started)
@oliver-goetz wdyt about this?